File: javascript/prompt_enhancement.js
/**
 * Prompt Enhancement — JavaScript
 *
 * Auto-rewrite vague prompts for better AI responses.
 * Works with both the native API and OpenAI SDK.
 *
 * Docs: https://routeplex.com/docs/prompt-enhancement
 * Usage: node prompt_enhancement.js
 */

import OpenAI from "openai";

const API_KEY = process.env.ROUTEPLEX_API_KEY;

if (!API_KEY) {
  console.error("Error: Set ROUTEPLEX_API_KEY environment variable");
  process.exit(1);
}

const client = new OpenAI({
  apiKey: API_KEY,
  baseURL: "https://api.routeplex.com/v1",
});

async function main() {
  // Enhancement via X-RoutePlex-Enhance header
  const response = await client.chat.completions.create(
    {
      model: "routeplex-ai",
      messages: [{ role: "user", content: "sort a list in python" }],
    },
    {
      headers: {
        "X-RoutePlex-Strategy": "quality",
        "X-RoutePlex-Enhance": "true",
      },
    }
  );

  console.log("Response:", response.choices[0].message.content?.slice(0, 200), "...");
  console.log(`Model: ${response.model}`);

  // Enhancement metadata in response (if available)
  if (response.x_routeplex?.enhancement) {
    const enh = response.x_routeplex.enhancement;
    console.log(`\nEnhancement applied: ${enh.applied}`);
    console.log(`Query type: ${enh.query_type}`);
  }

  // ── Standalone enhance endpoint (no API key required) ──────────────────
  console.log("\n--- Standalone Enhancement ---");
  const enhanceResp = await fetch("https://api.routeplex.com/api/v1/chat/enhance", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ prompt: "compare react and vue" }),
  });

  const enhanceData = await enhanceResp.json();
  console.log(`Original: ${enhanceData.data.original_prompt}`);
  console.log(`Enhanced: ${enhanceData.data.enhanced_prompt}`);
  console.log(`Type:     ${enhanceData.data.query_type}`);
}

main();

File: python/prompt_enhancement.py
"""
Prompt Enhancement
==================
Auto-rewrite vague prompts for better AI responses.
Enhancement is stateless, free, and adds no latency overhead.

Docs: https://routeplex.com/docs/prompt-enhancement
Usage: python prompt_enhancement.py
"""

import os
import requests

API_KEY = os.environ.get("ROUTEPLEX_API_KEY")
BASE_URL = "https://api.routeplex.com"

if not API_KEY:
    print("Error: Set ROUTEPLEX_API_KEY environment variable")
    print("  export ROUTEPLEX_API_KEY='your-api-key-here'")
    exit(1)


def main():
    # ── Chat with enhancement enabled ────────────────────────────────────────
    # A short, vague prompt gets automatically rewritten before hitting the model.
    response = requests.post(
        f"{BASE_URL}/api/v1/chat",
        headers={"Authorization": f"Bearer {API_KEY}"},
        json={
            "messages": [{"role": "user", "content": "sort a list in python"}],
            "mode": "routeplex-ai",
            "strategy": "quality",
            "enhance_prompt": True,
        },
    )

    if response.status_code != 200:
        print(f"Error {response.status_code}: {response.text}")
        exit(1)

    data = response.json()
    print("Response:", data["data"]["output"][:200], "...")
    print(f"Model used: {data['data']['model_used']}")

    # Enhancement metadata tells you what happened
    enhancement = data["data"].get("enhancement", {})
    print(f"\nEnhancement applied: {enhancement.get('applied')}")
    print(f"Original prompt: {enhancement.get('original_prompt')}")

    # ── Standalone enhance endpoint (no API key required) ────────────────────
    # Preview what RoutePlex will send to the model without making a chat call.
    print("\n--- Standalone Enhancement ---")
    enhance_resp = requests.post(
        f"{BASE_URL}/api/v1/chat/enhance",
        json={"prompt": "compare react and vue"},
    )

    enhance_data = enhance_resp.json()
    print(f"Original:  {enhance_data['data']['original_prompt']}")
    print(f"Enhanced:  {enhance_data['data']['enhanced_prompt']}")
    print(f"Changed:   {enhance_data['data']['changed']}")


if __name__ == "__main__":
    main()
